server/controllers/list.js

```diff
@@ -1135,39 +1142,50 @@ const updateListItem = (req, res) => {
         const data = { listId, userId, itemId, isOrdered };
 
         return returnPayload(
-          updateItem(socketInstance, dashboardClients, cohortClients)(data)
+          updateItem(
+            socketInstance,
+            dashboardClients,
+            cohortClients,
+            listClients,
+            viewersIds
+          )(data)
         )(doc);
       }
 
       if (name) {
         prevItemName = itemToUpdate.name;
         itemToUpdate.name = name;
         editedItemActivity = ActivityType.ITEM_EDIT_NAME;
-        itemToUpdate.editedBy = userId;
 
         const data = { listId, userId, itemId, name };
 
         return returnPayload(
-          updateItem(socketInstance, dashboardClients, cohortClients)(data)
+          updateItem(
+            socketInstance,
+            dashboardClients,
+            cohortClients,
+            listClients,
+            viewersIds
+          )(data)
         )(doc);
       }
 
       if (isArchived !== undefined) {
         itemToUpdate.isArchived = isArchived;
-        itemToUpdate.editedBy = userId;
-
-        if (isArchived) {
-          editedItemActivity = ActivityType.ITEM_ARCHIVE;
-        }
-
-        if (!isArchived) {
-          editedItemActivity = ActivityType.ITEM_RESTORE;
-        }
+        editedItemActivity = isArchived
+          ? ActivityType.ITEM_ARCHIVE
+          : ActivityType.ITEM_RESTORE;
 
         const data = { listId, userId, itemId, isArchived };
 
         return returnPayload(
-          updateItem(socketInstance, dashboardClients, cohortClients)(data)
+          updateItem(
+            socketInstance,
```

Comment:
hmm returnPayload is invoke 4 times (and only data is channging (I think?) Can we set Activity type and data inside `ifs` and call this metod once at the end of then?

Reply:
return payload is invoked only once on a single request. Data is different in every single if so I think it should stay as it is

Reply:
Yes, its invoked once per request but there is unnecessary code repetition I think that only data and activity type could be set in ifs. And after those ifs we could write the call of this function once and pass the data that was set before in ifs... If you think it is impossible -resolve this comment